ui/app/helpers/utils/conversion-util.js

```diff
@@ -60,60 +58,75 @@ const baseChange = {
-  pred,
-  R.converge(
-    conversionRateLens,
-    [R.pipe(R.prop(prop), setter), R.identity]
-  )
-)
-
-// conditional 'value' setting wrappers
-const whenPredSetWithPropAndSetter = (pred, prop, setter) => R.when(
-  pred,
-  R.converge(
-    valuePropertyLens,
-    [R.pipe(R.prop(prop), setter), R.identity]
-  )
-)
-const whenPropApplySetterMap = (prop, setterMap) => whenPredSetWithPropAndSetter(
-  R.prop(prop),
-  prop,
-  R.prop(R.__, setterMap)
-)
-
-// Conversion utility function
-const converter = R.pipe(
-  whenPredSetCRWithPropAndSetter(R.prop('conversionRate'), 'conversionRate', decToBigNumberViaString),
-  whenPredSetCRWithPropAndSetter(R.prop('invertConversionRate'), 'conversionRate', invertConversionRate),
-  whenPropApplySetterMap('fromNumericBase', toBigNumber),
-  whenPropApplySetterMap('fromDenomination', toNormalizedDenomination),
-  whenPredSetWithPropAndSetter(fromAndToCurrencyPropsNotEqual, 'conversionRate', convert),
-  whenPropApplySetterMap('toDenomination', toSpecifiedDenomination),
-  whenPredSetWithPropAndSetter(R.prop('numberOfDecimals'), 'numberOfDecimals', round),
-  whenPredSetWithPropAndSetter(R.prop('roundDown'), 'roundDown', roundDown),
-  whenPropApplySetterMap('toNumericBase', baseChange),
-  R.view(R.lensProp('value'))
-)
+/**
+ * Defines the base type of numeric value
+ * @typedef {('hex' | 'dec' | 'BN')} NumericBase
+ */
+
+/**
+ * Defines which type of denomination a value is in
+ * @typedef {('WEI' | 'GWEI' | 'ETH')} EthDenomination
+ */
+
+/**
+ * Utility method to convert a value between denominations, formats and currencies.
+ * @param {Object} input
+ * @param {string | BigNumber} input.value
+ * @param {NumericBase} input.fromNumericBase
+ * @param {EthDenomination} [input.fromDenomination]
+ * @param {string} [input.fromCurrency]
+ * @param {NumericBase} input.toNumericBase
+ * @param {EthDenomination} [input.toDenomination]
+ * @param {string} [input.toCurrency]
+ * @param {number} [input.numberOfDecimals]
+ * @param {number} [input.conversionRate]
+ * @param {boolean} [input.invertConversionRate]
+ * @param {string} [input.roundDown]
+ */
+const converter = ({
+  value,
+  fromNumericBase,
+  fromDenomination,
+  fromCurrency,
+  toNumericBase,
+  toDenomination,
+  toCurrency,
+  numberOfDecimals,
+  conversionRate,
+  invertConversionRate,
+  roundDown,
+}) => {
+  let convertedValue = toBigNumber[fromNumericBase ?? 'dec'](value)
+
+  if (fromDenomination) {
+    convertedValue = toNormalizedDenomination[fromDenomination](convertedValue)
+  }
+
+  if (fromCurrency !== toCurrency) {
+    let rate = toBigNumber.dec(conversionRate ?? 1)
+    if (invertConversionRate) {
+      rate = new BigNumber(1.0).div(conversionRate)
+    }
+    convertedValue = convertedValue.times(rate)
+  }
+
+  if (toDenomination) {
+    convertedValue = toSpecifiedDenomination[toDenomination](convertedValue)
+  }
+
+  if (numberOfDecimals) {
+    convertedValue = convertedValue.round(numberOfDecimals)
```

Comment:
This used the rounding mode `ROUND_HALF_DOWN`. The default is `ROUND_HALF_UP`, so this needs to be explicitly stated.
```suggestion
    convertedValue = convertedValue.round(numberOfDecimals, BigNumber.ROUND_HALF_DOWN)
```

Reply:
![image](https://user-images.githubusercontent.com/4448075/86943161-b6915400-c10b-11ea-8dd8-e90e37d62e62.png)
does this actually do anything?

Reply:
Ah, good point, I had missed that.

In that case, maybe this isn't required. Though it might at least protect against some dependency calling `BigNumber.config` and changing the global `BigNumber` config.

Reply:
Yeah, I think it's better to remove the snippet and declare it explicitly in the function call. Just threw me off because rounding worked appropriately in the tests.